rules/aip0122/http_url_pattern.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0122
+
+import (
+	"strings"
+
+	"github.com/googleapis/api-linter/lint"
+	"github.com/googleapis/api-linter/rules/internal/utils"
+	"github.com/jhump/protoreflect/desc"
+)
+
+// HTTP URL pattern shouldn't include underscore("_")
+var httpURLPattern = &lint.MethodRule{
+	Name:   lint.NewRuleName("core", "0122", "http-url-pattern"),
+	URI:    "https://aip.dev/122#guidance",
+	OnlyIf: isUnderscoreExisted,
```

Comment:
This technically works, but is not really the spirit. This really should not be here and have an `if hasUnderscore {` in `LintMethod`.

Reply:
Done